libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/CarAppLifecycleOwner.kt

```diff
@@ -34,34 +34,50 @@ internal class CarAppLifecycleOwner : LifecycleOwner {
     @VisibleForTesting
     internal val startedReferenceCounter = object : DefaultLifecycleObserver {
         override fun onCreate(owner: LifecycleOwner) {
-            lifecycleCreated++
-            logger.i(TAG, Message("LifecycleOwner ($owner) onCreate"))
-            if (activitiesCreated == 0 && lifecycleCreated > 0 && lifecycleForegrounded == 0) {
-                changeState(Lifecycle.State.STARTED)
+            if (createdChangingConfiguration > 0) {
+                createdChangingConfiguration--
+            } else {
+                lifecycleCreated++
+                logger.i(TAG, Message("LifecycleOwner ($owner) onCreate"))
+                if (activitiesCreated == 0 && lifecycleCreated > 0 && lifecycleForegrounded == 0) {
+                    changeState(Lifecycle.State.STARTED)
+                }
             }
         }
 
         override fun onStart(owner: LifecycleOwner) {
-            lifecycleForegrounded++
-            logger.i(TAG, Message("LifecycleOwner ($owner) onStart"))
-            if (activitiesCreated == 0 && lifecycleForegrounded > 0) {
-                changeState(Lifecycle.State.RESUMED)
+            if (foregroundedChangingConfiguration > 0) {
+                foregroundedChangingConfiguration--
+            } else {
+                lifecycleForegrounded++
+                logger.i(TAG, Message("LifecycleOwner ($owner) onStart"))
+                if (activitiesCreated == 0 && lifecycleForegrounded > 0) {
+                    changeState(Lifecycle.State.RESUMED)
+                }
             }
         }
 
         override fun onStop(owner: LifecycleOwner) {
-            lifecycleForegrounded--
-            logger.i(TAG, Message("LifecycleOwner ($owner) onStop"))
-            if (activitiesForegrounded == 0 && lifecycleForegrounded == 0) {
-                changeState(Lifecycle.State.STARTED)
+            if (owner is Activity && owner.isChangingConfigurations) {
```

Comment:
Can we also check for fragments something like this?
```kotlin
owner is Fragment && !owner.retainInstance && owner.activity?.isChangingConfigurations == true
```

Reply:
Ok I've added this check and now the `AppLifeycleActivity` uses an `ActiveGuidanceFragment` where you can test it out. Also there are unit tests to cover it.

One note, is that in order to reference the androidx.Fragment, I needed to add a dependency to the core library
`implementation dependenciesList.androidXFragment`